src/heaps/binomial_heap.jl

```diff
@@ -0,0 +1,371 @@
+# Binomial Heap
+mutable struct MutableBinomialHeapNode{T} 
+    data::T
+    degree::Int
+    handle::Int
+    parent::Union{MutableBinomialHeapNode,Nothing}
+    child::Union{MutableBinomialHeapNode,Nothing}
+    sibling::Union{MutableBinomialHeapNode,Nothing}
+end
+MutableBinomialHeapNode(data::T, handle::Int) where T = MutableBinomialHeapNode{T}(data, 0, handle, nothing, nothing, nothing)
+#################################################
+#
+#   heap type and constructors
+#
+#################################################
+mutable struct MutableBinomialHeap{T} <: AbstractHeap{T}
+    rootList::Vector{Union{Nothing,MutableBinomialHeapNode{T}}}
+    nodecount::Int
+    cumm_nodecount::Int
+    MutableBinomialHeap{T}() where T = new{T}([], 0, 0)
+
+    function MutableBinomialHeap{T}(xs::AbstractVector{T}) where T
+        BinomialHeap = _make_binomial_heap(xs, T)
+        new{T}(BinomialHeap.rootList, BinomialHeap.nodecount, BinomialHeap.cumm_nodecount)
+    end
+    
+end
+const BinomialHeap{T} = MutableBinomialHeap{T}
+BinomialHeap(xs::AbstractVector{T}) where T = MutableBinomialHeap{T}(xs)
+
+struct HeapBoundsError <: Exception
+    msg::AbstractString
+end
+
+function show(io::IO, _heap::MutableBinomialHeap{T}) where T
+    print(io, "MutableBinomialHeap")
+    list = _heap.rootList
+    n = length(list)
+    if n == 0
+        print(io, "{$(T)}")
+    end
+    print(io, "(")
+    for i = 1:n
+        print_tree(io, list[i])
+    end
+    print(io, ")")
+end
+
+function print_tree(io::IO, h::Union{MutableBinomialHeapNode{T},Nothing}) where T
+    while h !== nothing 
+        print(io, h.data)
+        print(io, ",")
+        print_tree(io, h.child) 
+        h = h.sibling; 
+    end
+end
+#################################################
+#
+#   core implementation
+#
+#################################################
+# make Binomial_heap from Vector
+function _make_binomial_heap(xs::AbstractVector{T}, Typ::Type{T}) where T
+    BinomialHeap = MutableBinomialHeap{Typ}()
+    for x in xs
+        push!(BinomialHeap, x)
+    end
+    BinomialHeap
+end
+
+# This function merge two Binomial Trees. 
+function merge_binomial_trees!(b1::MutableBinomialHeapNode{T}, b2::MutableBinomialHeapNode{T}) where T 
+    if b1.data > b2.data
+        b1, b2 = b2, b1
+    end
+    b2.parent = b1 
+    b2.sibling = b1.child 
+    b1.child = b2 
+    b1.degree += 1 
+    return b1
+end
+# adjust! function rearranges the heap so that 
+# heap is in increasing order of degree and 
+# no two binomial trees have same degree in this heap 
+function adjust!(heap::MutableBinomialHeap{T}) where T
```

Comment:
```suggestion
function adjust!(heap::MutableBinomialHeap)
```

`T` is never used so it does not need ot be in the type signature